### PR TITLE
Search: Remove require attribute on search inputs

### PR DESCRIPTION
**Pull request by jsnmoon**
Discovered while testing #20448.

#### Changes proposed in this Pull Request:
* Context: With WordPress 5.8, search widgets are automatically converted into search blocks which now output a search input with a newly added `required` attribute. This change prevents the user from spawning the Instant Search interface if the input is blank.
* This PR removes the `required` attribute when binding onto the search input.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* On a WordPress 5.8 installation with a Jetpack Search subscription, add a Search block to a page (or sidebar area).
* With Instant Search disabled, navigate to the page (or sidebar) with the Search block. Verify that the search input has the `required` attribute.
* Enable Instant Search via the Performance Settings page in wp-admin and repeat the above step. Ensure that the `required` attribute is no longer present.

### CODE REVIEW

**Review by kangzj — APPROVED** [review]
Good catch and well done ❤️ 
It works and looks good to me.

**Comment by jsnmoon** [comment]
Marked for inclusion in the 10.0 release, given that this bug can disrupt the Jetpack Search UX with the WordPress 5.8 update.

**Review by anomiex — APPROVED** [review]
Code looks sensible. I'll trust that @kangzj tested it.

**Comment by samiff** [comment]
cherry-picked to 10.0 release branch in bbb96eba5453eff82eb7918c24c8f2787153c892

File: projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx [hunk 1]
```diff
@@ -22,18 +22,22 @@ export default class DomEventHandler extends Component {
...
 		this.addEventListeners();
 	}
 
 	componentWillUnmount() {
 		this.removeEventListeners();
 	}
 
-	disableAutocompletion() {
+	disableUnnecessaryFormAndInputAttributes() {
+		// Disables the following attributes:
```
Comment (bluefuton) [inline projects/plugins/jetpack/modules/search/instant-search/components/dom-event-handler.jsx:34]: Thanks for explaining these 👍

**Comment by jsnmoon** [comment]
Thanks for merging, @samiff!

**Pull request merged**